clients/payment/src/App.tsx

```diff
@@ -1,35 +1,79 @@
+import { AsyncData, Result } from "@swan-io/boxed";
 import { ErrorBoundary } from "@swan-io/lake/src/components/ErrorBoundary";
 import { LoadingView } from "@swan-io/lake/src/components/LoadingView";
 import { colors } from "@swan-io/lake/src/constants/design";
+import { useUrqlQuery } from "@swan-io/lake/src/hooks/useUrqlQuery";
+import { isNullishOrEmpty } from "@swan-io/lake/src/utils/nullish";
 import { Suspense } from "react";
+import { StyleSheet } from "react-native";
 import { P, match } from "ts-pattern";
 import { Provider as ClientProvider } from "urql";
 import { ErrorView } from "./components/ErrorView";
+import { GetMerchantPaymentLinkDocument } from "./graphql/unauthenticated";
 import { NotFoundPage } from "./pages/NotFoundPage";
 import { PaymentPage } from "./pages/PaymentPage";
 import { SuccessPage } from "./pages/SuccessPage";
 import { logFrontendError } from "./utils/logger";
 import { Router } from "./utils/routes";
 import { unauthenticatedClient } from "./utils/urql";
 
-export const App = () => {
-  const route = Router.useRoute(["Success", "PaymentLink"]);
+const styles = StyleSheet.create({
+  base: {
+    backgroundColor: colors.gray[50],
+    flexGrow: 1,
+  },
+});
 
-  return (
-    <ErrorBoundary
-      key={route?.name}
-      onError={error => logFrontendError(error)}
-      fallback={({ error }) => <ErrorView error={error} />}
-    >
-      <Suspense fallback={<LoadingView color={colors.gray[100]} />}>
-        <ClientProvider value={unauthenticatedClient}>
-          {match(route)
-            .with({ name: "PaymentLink" }, () => <PaymentPage />)
-            .with({ name: "Success" }, () => <SuccessPage />)
-            .with(P.nullish, () => <NotFoundPage />)
-            .exhaustive()}
-        </ClientProvider>
-      </Suspense>
-    </ErrorBoundary>
+const UnauthenticatedClient = () => {
+  const route = Router.useRoute(["Form", "Success"]);
```

Comment:
we could have an `*Area` route with `paymentLinkId` and only run the following query when matched